xlators/mgmt/glusterd/src/glusterd.c

```diff
@@ -1395,8 +1395,8 @@ glusterd_destroy_hostname_list(glusterd_conf_t *priv)
 
     list_for_each_entry(hostname_obj, &priv->hostnames, hostname_list)
     {
-        GF_FREE(hostname_obj->hostname);
         list_del_init(&hostname_obj->hostname_list);
+        GF_FREE(hostname_obj->hostname);
```

Comment:
That's not the problem coverity has reported. The problem is that line 1400 is destroying `hostname_obj`, which will be needed in the next iteration of `list_for_each_entry()` to get the next item.

The correct way to fix this is to use `list_for_each_entry_safe()` or simply a `while (!list_empty(&priv->hostnames))`.